src/Sulu/Bundle/PreviewBundle/Resources/js/views/Preview/Preview.js

```diff
@@ -0,0 +1,265 @@
+            provider: resourceStore.resourceKey,
+            locale: locale,
+            id: resourceStore.id,
+        });
+        if (!route) {
+            return;
+        }
+
+        Requester.get(route).then((response) => {
+            this.setToken(response.token);
+        });
+
+        this.typeDisposer = autorun(() => {
+            if (formStore.loading || !this.iframeRef) {
+                return;
+            }
+
+            // $FlowFixMe
+            this.updateContext(formStore.type.get());
+        });
+
+        this.dataDisposer = autorun(() => {
+            if (resourceStore.loading || !this.iframeRef) {
+                return;
+            }
+
+            this.updatePreview(toJS(resourceStore.data));
+        });
+
+        this.started = true;
+    };
+
+    updatePreview = debounce((data) => {
+        const {
+            router: {
+                attributes: {
+                    locale,
+                    webspace,
+                },
+            },
+        } = this.props;
+
+        const route = previewConfigStore.generateRoute('update', {
+            locale: locale,
+            webspace: webspace,
+            token: this.token,
+        });
+        if (!route) {
+            return;
+        }
+
+        Requester.post(route, {data: data}).then((response) => {
+            const document = this.getPreviewDocument();
+            if (!document) {
+                return;
+            }
+
+            document.open();
+            document.write(response.content);
+            document.close();
+        });
+    }, previewConfigStore.debounceDelay);
+
+    updateContext = debounce((type) => {
+        const {
+            router: {
+                attributes: {
+                    webspace,
+                },
+            },
+        } = this.props;
+
+        const route = previewConfigStore.generateRoute('update-context', {
+            webspace: webspace,
+            token: this.token,
+        });
+        if (!route) {
+            return;
+        }
+
+        Requester.post(route, {context: {template: type}}).then((response) => {
+            const document = this.getPreviewDocument();
+            if (!document) {
+                return;
+            }
+
+            document.open();
+            document.write(response.content);
+            document.close();
+        });
+    }, previewConfigStore.debounceDelay);
+
+    componentWillUnmount() {
+        if (this.typeDisposer) {
+            this.typeDisposer();
+        }
+
+        if (this.dataDisposer) {
+            this.dataDisposer();
+        }
+
+        if (!this.started) {
+            return;
+        }
+
+        const route = previewConfigStore.generateRoute('stop', {token: this.token});
+        if (!route) {
+            return;
+        }
+
+        Requester.get(route);
+    }
+
+    getPreviewDocument = (): ?Document => {
+        // eslint-disable-next-line
```

Comment:
You can also add the name of the route after `eslint-disable-next-line`, then the other checks still apply